Telemetry/TelemetryMeasurements.swift

```diff
@@ -126,37 +126,47 @@ public class DistributionMeasurement: StaticTelemetryMeasurement {
 public class EventsMeasurement: TelemetryMeasurement {
     private let storage: TelemetryStorage
     private let pingType: String
-    
-    private var events: [[Any?]]
-    
+    private var eventsAddedToFile = 0
+
+    // Ensure at least this many events have been added before uploading.
     public var numberOfEvents: Int {
         get {
-            return events.count
+            return eventsAddedToFile
         }
     }
     
     init(storage: TelemetryStorage, pingType: String) {
         self.storage = storage
         self.pingType = pingType
-        
-        self.events = storage.get(valueFor: "\(pingType)-events") as? [[Any?]] ?? []
-        
         super.init(name: "events")
     }
     
     public func add(event: TelemetryEvent) {
-        events.append(event.toArray())
-
-        storage.set(key: "\(pingType)-events", value: events)
+        guard let data = event.toJSON() else {
+            return
+        }
+        eventsAddedToFile += 1
+        storage.appendEvent(data: data, forPingType: pingType)
```

Comment:
It might be a slightly cleaner API to pass the `TelemetryEvent` to this function directly instead of having to pass an `NSData` JSON blob. Then, have `appendEvent()` return a `Bool` indicating if it succeeded (so you can still increment the number of events added to the file).